File: sources/paopao/Kart/Physics/DriftControl.cs
using System;

public struct DriftControl
{
	public void Initialize()
	{
		this.slipMode = false;
		this.slipTime = 0f;
		this.forceSlip = false;
		this.trigger = false;
		this.triggerTime = 0f;
	}

	public override string ToString()
	{
		return string.Concat(new string[]
		{
			this.slipMode.ToString(),
			" ",
			this.slipTime.ToString(),
			" ",
			this.forceSlip.ToString(),
			" ",
			this.trigger.ToString(),
			" ",
			this.triggerTime.ToString()
		});
	}

	public bool slipMode;

	public float slipTime;

	public bool forceSlip;

	public bool trigger;

	public float triggerTime;
}

File: src/shared/kartEngine/KartMove/DriftControl.ts
/**
 * DriftControl struct的TypeScript实现
 * 与Lua版本完全一致，用于处理漂移控制状态
 */

export class DriftControl {
    public slipMode: boolean = false;
    public slipTime: number = 0.0;
    public forceSlip: boolean = false;
    public trigger: boolean = false;
    public triggerTime: number = 0.0;

    constructor() {
        this.Initialize();
    }

    public Initialize(): void {
        this.slipMode = false;
        this.slipTime = 0.0;
        this.forceSlip = false;
        this.trigger = false;
        this.triggerTime = 0.0;
    }

    // 静态工厂方法
    static create(): DriftControl {
        return new DriftControl();
    }
}

export default DriftControl;